tests/test_lightningd.py

```diff
@@ -51,19 +51,10 @@ def wait_for(success, timeout=30):
         raise ValueError("Error waiting for {}", success)
 
 
-def sync_blockheight(*args):
-    while True:
-        target = bitcoind.rpc.getblockcount()
-        all_up_to_date = True
-        for l in args:
-            if l.rpc.dev_blockheight()['blockheight'] != target:
-                all_up_to_date = False
-
-        if all_up_to_date:
-            return
-
-        # Sleep before spinning.
-        time.sleep(0.1)
+def sync_blockheight(nodes):
+    target = bitcoind.rpc.getblockcount()
+    for n in nodes:
+        wait_for(lambda: n.rpc.getinfo()['blockheight'] == target)
```

Comment:
`wait_for` receives a predicate in the form of a lambda expression that is evaluated multiple times until it returns `true`. It also wraps the various things such as timeout interval and polling interval in a single function, but delegates the actual success evaluation to the passed in lambda.

    lambda: evaluated_value == expected_value

is a lambda expression, without argument, that returns a boolean indicating a match or not :wink: